### PR TITLE
docs(contrib): Suggest atomic commits with separate test commits

### PR DESCRIPTION
**Pull request by epage**
### What does this PR try to resolve?

This came up in a discussion with @torhovland.

### How should we test and review this PR?

I'm unsure what the right weight is to put behind these suggestions
- I don't want a lack of git history editing experience to get in the way of someone contributing
- If people do anything, I'd like to see the tests split out

### CODE REVIEW

File: src/doc/contrib/src/process/working-on-cargo.md [hunk 1]
```diff
@@ -124,6 +124,16 @@ typically finish in under 30 minutes.
 The reviewer might point out changes deemed necessary. Large or tricky changes
 may require several passes of review and changes.
 
+> **tip:** Prefer atomic commits where each commit is a single, complete, and coherent unit of work.
+> For example, if your feature work leads to renaming a module, make the rename its own commit.
+> However, adding an internal function that is unused is not complete of coherent.
+>
+> As part of your atomic commits, prefer adding tests as their own commit *before* any functionality changes.
```
Comment (weihanglo) [inline src/doc/contrib/src/process/working-on-cargo.md:131]: Could we have an example PR here?

Some people write a failing test first, not a test passing the CI.

Comment (epage) [reply src/doc/contrib/src/process/working-on-cargo.md:131]: Added examples and tried to be clearer that we want the tests passing on each commit.

**Review by weihanglo — APPROVED** [review]
It looks reasonable as a tip not an enforcement.
Also it is something we can copy over and link to when communicating with contributors.

**Comment by epage** [comment]
Went ahead and remove draft status.  Started it as one to be very intentional about merging this in but based on @weihanglo's comment, seems like this is good to go forward.

**Comment by weihanglo** [comment]
@bors r+

Thanks!

**Pull request merged**